Sources/MapboxMaps/Location/Puck/PuckType.swift

```diff
@@ -76,6 +80,35 @@ public struct Puck2DConfiguration: Equatable {
         self.showsAccuracyRing = showsAccuracyRing
         self.accuracyRingColor = accuracyRingColor
         self.accuracyRingBorderColor = accuracyRingBorderColor
+        self.puckBearingEnabled = true
+    }
+
+    /// Initialize a `Puck2D` object with a top image, bearing image, shadow image, scale, and accuracy ring visibility.
+    /// - Parameters:
+    ///   - topImage: The image to use as the top layer for the location indicator.
+    ///   - bearingImage: The image used as the middle of the location indicator.
+    ///   - shadowImage: The image that acts as a background of the location indicator.
+    ///   - scale: The size of the images, as a scale factor applied to the size of the specified image..
+    ///   - showsAccuracyRing: Indicates whether the location accurary ring should be shown.
+    ///   - accuracyRingColor:The color of the accuracy ring.
+    ///   - accuracyRingBorderColor: The color of the accuracy ring border.
+    ///   - puckBearingEnabled: Indicates whether the puck rotates to track the bearing source.
+    public init(topImage: UIImage? = nil,
+                bearingImage: UIImage? = nil,
+                shadowImage: UIImage? = nil,
+                scale: Value<Double>? = nil,
+                showsAccuracyRing: Bool = false,
+                accuracyRingColor: UIColor = UIColor(red: 0.537, green: 0.812, blue: 0.941, alpha: 0.3),
+                accuracyRingBorderColor: UIColor = UIColor(red: 0.537, green: 0.812, blue: 0.941, alpha: 0.3),
+                puckBearingEnabled: Bool = true) {
+        self.topImage = topImage
+        self.bearingImage = bearingImage
+        self.shadowImage = shadowImage
+        self.scale = scale
+        self.showsAccuracyRing = showsAccuracyRing
+        self.accuracyRingColor = accuracyRingColor
+        self.accuracyRingBorderColor = accuracyRingBorderColor
+        self.puckBearingEnabled = puckBearingEnabled
     }
 
     /// Create a Puck2DConfiguration instance with or without an arrow bearing image. Default without the arrow bearing image.
```

Comment:
Now it will make sense to disable 2D puck bearing in the default configuration in case `showBearing` is `false`. As bearing updates won't be visible anyway with the dot image.
```diff
    public static func makeDefault(showBearing: Bool = false) -> Puck2DConfiguration {
        let shadowImage = UIImage(named: "location-dot-outer", in: .mapboxMaps, compatibleWith: nil)!
        return Puck2DConfiguration(
            topImage: UIImage(named: "location-dot-inner", in: .mapboxMaps, compatibleWith: nil)!,
            bearingImage: showBearing ? makeBearingImage(size: shadowImage.size) : nil,
-           shadowImage: shadowImage)
            shadowImage: shadowImage,
+           puckBearingEnabled: showBearing)
    }
```